Sources/_StringProcessing/StringExtras.swift

```diff
@@ -0,0 +1,143 @@
+
```

Comment:
For a new file, we need to:
- [ ]  include the Swift.org preamble 
- [ ] add it to `CMakeLists.txt`